fingerprint.c

```diff
@@ -66,6 +99,10 @@ void *enroll_thread_loop(void *arg)
     while((status = fpc_capture_image(sdev->fpc)) >= 0) {
         ALOGD("%s : Got Input status=%d", __func__, status);
 
+        if (getState(sdev) != STATE_ENROLL) {
+            break;
```

Comment:
this is more for when a cancelled state has been set outside of the thread (by android) and we are still within the enroll "wait for image loop" without this check it will be unable to break out of this loop

Reply:
This is to check the current state within the enroll loop, without this there is no way of jumping out of the loop "while((status = fpc_capture_image(sdev->fpc)) >= 0) {" if android called cancel.